### PR TITLE
Add parallel hedging sample

### PR DESCRIPTION
**Pull request by peter-csala**
# Addressed issue
- #63 

# Description
- Added `Demo15_ParallelHedging` to demonstrate the parallel mode
- Fixed a bug in `Demo14_FallbackHedging-RetryWithFallback`
- Added a _How to read the demo logs_ section to each hedging demos' summary

### CODE REVIEW

File: PollyTestWebApi/Program.cs [hunk 1]
```diff
@@ -25,7 +25,7 @@
 app.UseRateLimiter();
 
 // Register two endpoints that are not rate limited.
-// They are used by the demo 10 and 11 (concurrency limiter)
+// They are used by demo 10 and 11 (concurrency limiter)
```
Comment (martincostello) [inline PollyTestWebApi/Program.cs:28]:
```suggestion
// They are used by demos 10 and 11 (concurrency limiter)
```

File: PollyTestWebApi/Program.cs [hunk 2]
```diff
@@ -47,11 +47,16 @@
...
 // Register a cancellable endpoint that is not rate limited.
-// It is used by demos 13 and 14 (hedging)
-app.MapGet("/api/VaryingResponseStatus/{id}", async (CancellationToken token, [FromRoute] string id) =>
+// It is used by demos 13, 14 and 15 (hedging)
+app.MapGet("/api/VaryingResponseStatus/{id}", async (CancellationToken token, [FromRoute] string id, [FromQuery] bool? useJitter) =>
 {
+    var jitter = Random.Shared.Next(-200, 200);
+    var delay = useJitter == true
+     ? TimeSpan.FromSeconds(0.5)
+     : TimeSpan.FromSeconds(0.5) + TimeSpan.FromMilliseconds(jitter);
```
Comment (martincostello) [inline PollyTestWebApi/Program.cs:56]: I think you have your switch cases the wrong way around.

Comment (peter-csala) [reply PollyTestWebApi/Program.cs:56]: You have eagle eyes :D

File: PollyTestWebApi/Program.cs [hunk 3]
```diff
@@ -51,7 +51,7 @@
 app.MapGet("/api/VaryingResponseStatus/{id}", async (CancellationToken token, [FromRoute] string id, [FromQuery] bool? useJitter) =>
 {
     var jitter = Random.Shared.Next(-200, 200);
-    var delay = useJitter == true
+    var delay = useJitter == false
```
Comment (martincostello) [inline PollyTestWebApi/Program.cs:54]: Is it intentional that `useJitter` being `null` uses jitter?

Comment (peter-csala) [reply PollyTestWebApi/Program.cs]: Fixed it, thanks again

File: PollyTestWebApi/Program.cs [hunk 4]
```diff
@@ -51,9 +51,11 @@
 app.MapGet("/api/VaryingResponseStatus/{id}", async (CancellationToken token, [FromRoute] string id, [FromQuery] bool? useJitter) =>
 {
     var jitter = Random.Shared.Next(-200, 200);
-    var delay = useJitter == false
-     ? TimeSpan.FromSeconds(0.5)
-     : TimeSpan.FromSeconds(0.5) + TimeSpan.FromMilliseconds(jitter);
+    var delay = TimeSpan.FromSeconds(0.5);
+    if(useJitter == true)
```
Comment (martincostello) [inline PollyTestWebApi/Program.cs:55]:
```suggestion
    if (useJitter == true)
```

**Review by martincostello — APPROVED** [review]

**Pull request merged**